src/steps/build-account-to-member-mapping/index.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  Entity,
+  Relationship,
+  createIntegrationRelationship,
+  JobState,
+} from '@jupiterone/integration-sdk';
+import {
+  STEP_ID as MEMBER_STEP,
+  ACCOUNT_MEMBER_TYPE,
+} from '../fetch-enterprise-account-members';
+import {
+  STEP_ID as ACCOUNT_STEP,
+  ACCOUNT_TYPE,
+} from '../fetch-enterprise-accounts';
+
+const step: IntegrationStep = {
+  id: 'build-account-to-member-relationships',
+  name: 'Build Account-to-Member Relationships',
+  types: [],
+  dependsOn: [ACCOUNT_STEP, MEMBER_STEP],
+  async executionHandler({ jobState }: IntegrationStepExecutionContext) {
+    const accountIdMap = await createAccountIdMap(jobState);
+
+    await jobState.iterateEntities(
+      { _type: ACCOUNT_MEMBER_TYPE },
+      async (member) => {
+        const account = accountIdMap.get(member.enterpriseAccountId as string);
+
+        if (account) {
+          await jobState.addRelationships([
+            createAccountToMemberRelationship(account, member),
+          ]);
+        }
+      },
+    );
+  },
+};
+
+export default step;
+
+async function createAccountIdMap(
+  jobState: JobState,
+): Promise<Map<string, Entity>> {
+  const accountIdMap = new Map<string, Entity>();
+  await jobState.iterateEntities({ _type: ACCOUNT_TYPE }, (account) => {
+    // unfortunately need to cast because of EntityPropertyValue type
```

Comment:
Instead of casting, you should define a type for your entities that extends the SDK's `Entity` type. This would be much safer.

Reply:
https://github.com/JupiterOne/data-model/blob/master/src/schemas/Entity.json#L12 says it should be a string, but that it is optional on entities. I think it's okay to use `account.id as string` here instead of creating a sub-type, which then propagates throughout the rest of the code even though most things aren't going to reference the `id` property. In fact, I'm expecting this whole map building thing to go away with https://github.com/JupiterOne/integration-sdk/issues/79.

Reply:
What about something like 
```typescript
async function createAccountIdMap(
  jobState: JobState,
): Promise<Map<string, AccountEntity>> {
  const accountIdMap = new Map<string, AccountEntity>();
  await jobState.iterateEntities({ _type: ACCOUNT_TYPE }, (entity) => {
    const account = entity as AccountEntity;
    accountIdMap.set(account.id, account);
  });
  return accountIdMap;
}
```
Worth the change?

Reply:
IMO, no. It's just trading one typecast for another, except now there has to be an `AccountEntity` type as well. That said, `iterateEntities` should probably become `iterateEntities<T extends Entity>`, and we need `interface Entity = { ..., id?: string }` (generated from the schemas, as part of the `data-model` project).

Reply:
Actually, you could simply `await jobState.iterateEntities({ _type: ACCOUNT_TYPE }, (entity: AccountEntity) => {` and there would be no need for a cast. I agree with what @aiwilliams has said though and I look forward to seeing the map building code go away with SDK improvements.

Reply:
I'm also looking forward to the SDK improvements. For now, is this one good to go? 

Also, when those SDK improvements _do_ arrive, will our strategy include going back to refactor a lot of this code?

Reply:
Yeah, I think this is all good to go! I think this is why a CHANGELOG is so important: when upgrading the SDK package, check out the CHANGELOG and adopt the new patterns.